### DISCUSSION TITLE
Whats the preferred technique to pass multiple "parameters" to an alpine enabled blade component?

### DISCUSSION THREAD

**Discussion by brino** [alpinejs/alpine#4351] [1. Help]
I've tried using x-modelable="params" and x-model="{foo:['array'],bar:'string'}" to pass a "params" object ... it doesn't work I get the following error.

Using Laravel 11, with blade and alpine.

> app-aef45551.js:6 Alpine Expression Error: Invalid destructuring assignment target
> 
> Expression: "{
>                                         recipients: message.recipients,
>                                         parts: message.parts,
>                                         subject: 'RE: '+message.subject,
>                                         body: message.subcomments
>                                     } = __placeholder"

Any input or direction would be greatly appreciated.

**Comment by brino** [comment]
I get the impression the answer is going to be simply, "Alpine doesn't do that"... but I'm **hoping** someone out there has figured this out!

**Comment by ekwoka** [comment]
You'd need to actually have the object stored.

Or better yet, use wire entangle.

**Reply by brino** [reply]
Stored where, the DB? Or like, window.params??

And I'd really like to avoid using a secondary lib (livewire) but I'll check it out.

Thanks!

**Reply by ekwoka** [reply]
I mean the thing you have model on. You need that object to exist at a property in the stack.

Whatever you put in x-model is used as a set and a get. So if x-modelable is `foo` and x-model is `bar` imagine it like writing `foo = bar` and `bar = foo`

If you write a new object `{ fizz: 'buzz' }` then you get `bar = { fizz: 'buzz' }` and `{ fizz: 'buzz' } = bar`. The first one works, but the second doesn't.
